katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+
+
+class H5DataV2_5(DataSet):
+    """Load HDF5 format version 2.5 file produced by AVN DBE.
+
+    For more information on attributes, see the :class:`DataSet` docstring.
+
+    Parameters
+    ----------
+    filename : string
+        Name of HDF5 file
+    ref_ant : string, optional
+        Name of reference antenna, used to partition data set into scans
+        (default is first antenna in use)
+    time_offset : float, optional
+        Offset to add to all timestamps, in seconds
+    mode : string, optional
+        HDF5 file opening mode (e.g. 'r+' to open file in write mode)
+    quicklook : {False, True}
+        True if synthesised timestamps should be used to partition data set even
+        if real timestamps are irregular, thereby avoiding the slow loading of
+        real timestamps at the cost of slightly inaccurate label borders
+    keepdims : {False, True}, optional
+        Force vis / weights / flags to be 3-dimensional, regardless of selection
+    kwargs : dict, optional
+        Extra keyword arguments, typically meant for other formats and ignored
+
+    Attributes
+    ----------
+    file : :class:`h5py.File` object
+        Underlying HDF5 file, exposed via :mod:`h5py` interface
+    """
+    def __init__(self, filename, ref_ant='', time_offset=0.0, mode='r', quicklook=False, keepdims=False, **kwargs):
+        DataSet.__init__(self, filename, ref_ant, time_offset)
+
+        # Load file
+        self.file, self.version = H5DataV2_5._open(filename, mode)
+        f = self.file
+
+        # Load main HDF5 groups
+        data_group    = f["Data"]
+        config_group  = f["MetaData/Configuration"]
+        sensors_group = f["MetaData/Sensors"]
+        markup_group  = f["Markup"]
+
+        # Get observation script parameters, with defaults
+        for k, v in config_group['Observation'].attrs.iteritems():
+            self.obs_params[str(k)] = v
+        self.observer      = self.obs_params.get('observer', '')
+        self.description   = self.obs_params.get('description', '')
+        self.experiment_id = self.obs_params.get('experiment_id', '')
+
+        # ------ Extract timestamps ------
+        accumulation_length = get_single_value(config_group['DBE'], 'accum_length')  # Accumulation length in number of FPGA frames.
+        coarse_size = get_single_value(config_group["DBE"], "dbe.fft.coarse.size")
+        fine_size = get_single_value(config_group["DBE"], "dbe.fft.fine.size")
+        sampling_frequency = 800e6  # Have to hardcode this for the time being.
+        self.dump_period  = float(accumulation_length) * coarse_size / sampling_frequency
+        if fine_size != 0:
+            self.dump_period *= fine_size
+
+        # Obtain visibility data and timestamps
+        self._vis         = data_group['VisData'][:-1,:,:]
+        self._stokes      = data_group['StokesData']
+        self._timestamps  = data_group['Timestamps']
+        self._time_av_ll  = data_group["Left Power time average"]
```

Comment:
Out of curiosity, how are these averages calculated?

Reply:
The ingest software just averages across the entire spectrum for each dump, keeps it in memory and writes the entire vector as an H5 dataset at the same time as it's writing timestamps, etc. It's just a rough-and-ready piece of data to use for quick checks of the H5 file if you don't have KatDAL handy.